Editor+

Rimmuru opened the pull request:
feat(world): add train
feat(train): add hijack train
feat(train): add train speed
feat(train): add delete train
feat(train): add derail train
feat(train): add drive train
feat(train): add exit train
feat(Troll): add set bounty cmd

Rimmuru committed:
* feat(Toxic): add send to island

Rimmuru committed:
Merge branch 'YimMenu:master' into master

Rimmuru committed:
feat(mobile): add helicopter backup
refractor(mobile): update seperator

Rimmuru committed:
Merge branch 'master' of https://github.com/Rimmuru/YimMenu

Rimmuru committed:
feat(Troll): add set bounty

Rimmuru committed:
bounty actually works now

Rimmuru committed:
fix an oopsie

Rimmuru committed:
remove useless 3rd param

Rimmuru committed:
tps
tp all to island
tp all to eclipse
tp to eclipse

Yimura commented on src/util/troll.hpp:
```diff
@@ -0,0 +1,40 @@
+            (int64_t)eRemoteEvent::Bounty,    
+            target->id(),
+            1,
+            10000,
```
10k bounty is a modder only bounty value, allow for setting a custom value possibly.

Also weird indents.

Rimmuru replied:
done the value btws, the indents is the default vs tab indent

Rimmuru committed:
add bounty value

Yimura reviewed (dismissed):
Looks good, can be merged into `master`.

Edit:
nvm, just saw the comment from @maybegreat48

tupoy-ya commented on src/util/troll.hpp:
```diff
@@ -0,0 +1,41 @@
+
+namespace big::troll
+{
+	inline void set_bounty_on_player(player_ptr target, int value)
```
Would be nice to have it as a command too.

Loko415 replied:
very easy to do lmao

Loko415 replied:
the commands system is 10/10 and its pretty easy to convert stuff into commands

Yimura replied:
@Loko415 you still have got a problem with posting stuff into multiple comments for something that could've been edited in.

Rimmuru committed:
compliance
removed gui for some that are cmds
added bounty command
add set bounty cmd
add train
another spam text

Rimmuru committed:
working hijack &, delete. + placeholders

Rimmuru committed:
speed, derail

Rimmuru committed:
Merge branch 'YimMenu:master' into master

Rimmuru committed:
drive train, working derail

Rimmuru committed:
Merge branch 'master' of https://github.com/Rimmuru/YimMenu

Rimmuru committed:
improved drive train

Rimmuru committed:
exit train, rework delete

Rimmuru committed:
im an idiot

Rimmuru committed:
resolve conflicts (i think)

Rimmuru committed:
.

Yimura commented on src/util/train.hpp:
```diff
@@ -0,0 +1,69 @@
+			}
+		}
+		return result;
+	};
```
Why is index[0] set to `300`?
Why does the iterator start at 1?

`std::vector::size()` is a method returns the amount of entries in a vector.

Try taking a look at how the replay interface works, you can easily iterate of all the entities that exist on your local machine and push them onto a vector.

Rimmuru replied:
that function was written by one the guys above

Yimura replied:
I'll take a look at replacing it with the replay interface.

Rimmuru replied:
better?

Yimura commented on src/views/self/view_mobile.cpp:
```diff
@@ -7,26 +7,42 @@ namespace big
 		components::button("Request Ammo Drop", [] {
 			mobile::merry_weather::request_ammo_drop();
-		});
+			});
```
Indent

Yimura commented on src/views/self/view_mobile.cpp:
```diff
@@ -7,26 +7,42 @@ namespace big
 		components::button("Helicopter Pickup", [] {
 			mobile::merry_weather::request_helicopter_pickup();
-		});
+			});
```
Indent

Yimura commented on src/views/self/view_mobile.cpp:
```diff
@@ -7,26 +7,42 @@ namespace big
+
+		components::button("Request Backup Helicopter", [] {
+			mobile::merry_weather::request_backup_helicopter();
+			});
```
Indent

Yimura commented on src/views/self/view_mobile.cpp:
```diff
@@ -7,26 +7,42 @@ namespace big
+
+		components::button("Request Airstrike", [] {
+			mobile::merry_weather::request_airstrike();
+			});
```
Indent

Yimura commented on src/views/self/view_mobile.cpp:
```diff
@@ -7,26 +7,42 @@ namespace big
+
+		components::button("Bullshark Testosterone", [] {
+			mobile::ceo_abilities::request_bullshark_testosterone();
+			});
```
Indent

Yimura commented on src/views/world/view_train.cpp:
```diff
@@ -0,0 +1,53 @@
+#include "services/model_preview/model_preview_service.hpp"
+#include "services/players/player_service.hpp"
+
+#include <imgui_internal.h>
```
Are all  these includes required?

Yimura commented on src/views/world/view_train.cpp:
```diff
@@ -0,0 +1,53 @@
+		ImGui::SameLine();
+		
+		components::button("Delete Train", []
+			{
+				train::delete_train();
+			});
```
Indent

Yimura commented on src/views/world/view_train.cpp:
```diff
@@ -0,0 +1,53 @@
+	void view::train()
+	{
+		components::button("Hijack Train", []
+			{
+				train::hijack_train();
+			});
```
Indent

Yimura commented on src/views/world/view_train.cpp:
```diff
@@ -0,0 +1,53 @@
+		ImGui::SameLine();
+		
+		components::button("Exit Train", []
+			{
+				train::exit_train();
+			});
```
Indent

Rimmuru committed:
includes, indents
(git pls doesnt break it)

Rimmuru committed:
conflicts

Rimmuru committed:
Merge branch 'master' into master

Rimmuru committed:
using CReplayInterface
hijack seems to have broken tho

Rimmuru committed:
Merge branch 'master' of https://github.com/Rimmuru/YimMenu

Rimmuru committed:
improvement i guess

Rimmuru committed:
Merge branch 'master' into master

Yimura commented on src/util/train.hpp:
```diff
@@ -0,0 +1,80 @@
+			while (!NETWORK::NETWORK_HAS_CONTROL_OF_ENTITY(train))
+			{
+				script::get_current()->yield();
+			}
```
Could end up infinitely looping causing the fiber pool to lockup, make use of the `entity::request_control_of()` function which has a built in breakout if it fails to get control of the entity.

Rimmuru replied:
assuming you meant entity::take_control_of, switched it over

Yimura replied:
Yup, seems correct now.

Aure7138 commented on src/util/train.hpp:
```diff
@@ -0,0 +1,80 @@
+	{
+		std::vector<Vehicle> result;
+		rage::CReplayInterface* CReplayInterface_var = *g_pointers->m_replay_interface; 
+		for (int i = 0; i <= 300; i++)
```
< 300 would be better, <= 300 is actually 301 cars

Rimmuru replied:
agreed, sorted

Aure7138 commented on src/core/globals.hpp:
```diff
@@ -387,6 +387,12 @@ namespace big
 			NLOHMANN_DEFINE_TYPE_INTRUSIVE(spawn_ped, preview_ped)
 		} spawn_ped{};
 
+		struct train
+		{
+			bool derail_train = false;
+			bool drive_train = false;
+		} train{};
```
Just a reminder, you can use NLOHMANN_DEFINE_TYPE_INTRUSIVE to save settings, the code looks good to me

Rimmuru replied:
the idea with them though was they wouldnt be saved

Aure7138 replied:
ah, alright

Rimmuru replied:
:)

Rimmuru committed:
control fix?

Rimmuru committed:
Merge branch 'master' of https://github.com/Rimmuru/YimMenu

Rimmuru committed:
oops
